xtask/src/main.rs

```diff
@@ -214,6 +214,8 @@ fn qemu(profile: Build) -> Result<()> {
         .arg("-nographic")
         .arg("-cpu")
         .arg("kvm64,+rdtscp,+pdpe1gb,+fsgsbase,+x2apic")
+        .arg("-m")
+        .arg("2048")
```

Comment:
Is this safe given we're mapping the first 4 GiB of memory for use?

Reply:
Yes. Prior to this, it allocated much less (like 256 megs or something) and I was actually writing off the end of RAM loading the binaries. Indeed, I need to add some more error checking here.

Reply:
I mean, though, won't we still be at risk of writing off the end of memory if we're mapping up to 4GB but only exposing a machine with 2GB?  (at least, without checking)

Reply:
Oh, and a quick note on the 4GiB thing: there are things in the physical address space that aren't necessarily RAM (like the PCI hole). Mapping those 4GiB lets us move out of 32-bit protected mode with everything still conveniently accessible in the same way it was before turning on paging, and it happens to be reasonably convenient in theon as well.

Reply:
Added some more checking here.

(I also didn't realize we were colliding on comments! Too funny.)